coderd/agentapi/metadata.go

```diff
@@ -107,6 +108,9 @@ func (a *MetadataAPI) BatchUpdateMetadata(ctx context.Context, req *agentproto.B
 		)
 	}
 
+	// Inject RBAC object into context for dbauthz fast path, avoid having to
+	// call GetWorkspaceByAgentID on every metadata update.
+	ctx = a.RBACContextFn(ctx)
 	err = a.Database.UpdateWorkspaceAgentMetadata(ctx, dbUpdate)
```

Comment:
In case `ctx` is used in the function later, we should require the cache to be populated per request.
```suggestion
	err = a.Database.UpdateWorkspaceAgentMetadata(a.RBACContextFn(ctx), dbUpdate)
```